### ISSUE TITLE
Cannot reproduce the results in the paper

### ISSUE THREAD

**Issue by ChocoOreal** [kakaobrain/cxr-clip#8]
I tried training the pretrain model on mimic dataset with 30 epochs, batch size 32, lr 5e-6, weight decay 1e-4, the optimizer I used was adamw. Other configs were analogous to the default configs in the repo. After completing 30 epochs, this is the loss I got over each epoch: 
![image](https://github.com/kakaobrain/cxr-clip/assets/83420027/bee05c0f-2371-41a5-a879-3294a50a0bb7)
When I run evaluate_clip script, I got this result: 
![image](https://github.com/kakaobrain/cxr-clip/assets/83420027/2caec32b-a5d6-47eb-b593-d0d1f9519b87)
which is inferior to your results.

This is my data after processed:
```bash
subject_id,study_id,image,view,PA,LATERAL,AP,LL,nan,LAO,RAO,AP AXIAL,SWIMMERS,PA LLD,AP LLD,XTABLE LATERAL,AP RLD,PA RLD,LPO,text,split,text_augment
10000032,53189527,"['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53189527/2a2277a9-b0ded155-c0de8eb9-c124d10e-82c5caab.jpg', '/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53189527/e084de3b-be89b11e-20fe3f9f-9c8d8dfe-4cfd202c.jpg']","['LATERAL', 'PA']",['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53189527/2a2277a9-b0ded155-c0de8eb9-c124d10e-82c5caab.jpg'],['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53189527/e084de3b-be89b11e-20fe3f9f-9c8d8dfe-4cfd202c.jpg'],,,,,,,,,,,,,,"['The cardiac, mediastinal and hilar contours are normal. Pulmonary vasculature\n is normal.  Lungs are clear. No pleural effusion or pneumothorax is present.\n Multiple clips are again seen projecting over the left breast.  Remote\n left-sided rib fractures are also re- demonstrated.', 'No acute cardiopulmonary abnormality.']",train,"['Heart contours, medistina and hilar are normal. Lung vascularity and usual. Lung. Lung is clear. No pleural effusion or pneumometers. Multiple clips are still being seen projecting to the left part of the breast. They are also remote fractures of the left ribs are re-proved.', 'No acute cardiac pulmonary disease and dysrhythmia.']"
10000032,53911762,"['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53911762/68b5c4b1-227d0485-9cc38c3f-7b84ab51-4b472714.jpg', '/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53911762/fffabebf-74fd3a1f-673b6b41-96ec0ac9-2ab69818.jpg']",['AP'],,,"['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53911762/68b5c4b1-227d0485-9cc38c3f-7b84ab51-4b472714.jpg', '/mnt/ssd1/CXR/data/imgs/p10/p10000032/s53911762/fffabebf-74fd3a1f-673b6b41-96ec0ac9-2ab69818.jpg']",,,,,,,,,,,,,"['Single frontal view of the chest provided.\n \n There is no focal consolidation, effusion, or pneumothorax. The\n cardiomediastinal silhouette is normal.  Again seen are multiple clips\n projecting over the left breast and remote left-sided rib fractures.  No free\n air below the right hemidiaphragm is seen.', 'No acute intrathoracic process.']",train,"['One front view of your chest does not make a focal effect, effusion, or pneumothorax. cardiomediastinal outlines are normally common to one another. Again, it can be seen that you are projecting several clips onto your left chest and off the fractures of your left side coast. No free air under your right hemidiafragma should be observed.', ""Yeah, well, it can't really be discreet.""]"
10001122,53447138,"['/mnt/ssd1/CXR/data/imgs/p10/p10001122/s53447138/8039752c-2ea661b7-16f1eafe-055b7e7b-dbd4cdd1.jpg', '/mnt/ssd1/CXR/data/imgs/p10/p10001122/s53447138/832b57d8-3ae08663-e152699e-51c5db98-b7cb4226.jpg']",[],,,,,,,,,,,,,,,,['The lung volumes are normal.  No evidence of TB or other parenchymal changes. \n Mild elevation of the left hemidiaphragm.  No pleural effusions.  No\n pneumonia.  The lateral radiograph shows evidence of anterior ligament\n calcification at the anterior aspect of the thoracic spine.  Status post\n cholecystectomy.'],train,"['There is no sign that there are other parenchymal changes and TB, no pleural effusion. Lateral X-ray shows evidence of calciification of ligament at the rear of the front of the rib. Post cholecystitomy. No sign are observed of TB and any other parenchymal variations.']"
10000032,56699142,['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s56699142/ea030e7a-2e3b1346-bc518786-7a8fd698-f673b44c.jpg'],['AP'],,,['/mnt/ssd1/CXR/data/imgs/p10/p10000032/s56699142/ea030e7a-2e3b1346-bc518786-7a8fd698-f673b44c.jpg'],,,,,,,,,,,,,"['The lungs are clear of focal consolidation, pleural effusion or pneumothorax. \n The heart size is normal.  The mediastinal contours are normal. Multiple\n surgical clips project over the left breast, and old left rib fractures are\n noted.', 'No acute cardiopulmonary process.']",train,"['They are free from focal bloated moments, pleural precipitations or pneumothorax, normal heart size. The thoracic contours are normal. There are numerous surgical fractures in the left breast and the old fractures of left ribs have been noted.', 'No acute cardio-pilmonerous processes.']"
```
Can you elaborate if my processed data is wrong somewhere.  I wonder if keeping other views than AP, PA, and LATERAL leads to the inferior figures. Plus, can you share me your training logs?

I look forward to you supports. Thank you in advance.

**Comment by kihyunU** [comment]
As you know, the contrastive learning is highly dependent on the batch size, so we recommend trying with a larger batch size.

**Comment by ChocoOreal** [comment]
I tried to evaluate I2T retrievale with the official weight but the results are still not comparable to the figures in your paper. Here is the results I got: 
![image](https://github.com/kakaobrain/cxr-clip/assets/83420027/8baa2fec-2534-4b87-9a4a-b46644008d7f)
Can you share more details on the data used for evaluation?
